Adaptation so that the server process also runs under Windows

JanNiklasWeder opened the pull request:
NEW: there is a container for the server that can be used from outside so that the server works directly in its own process and thus runs independently of the main process. This container takes care of providing some information about the server, such as its port and IP address.

UPDATED: The GitHub action for the tests now runs on every pull request and without the need for syntax checks.

UPDATED; The tests for the server have been adjusted so that they work with the other changes.

JanNiklasWeder committed:
Adaptation so that the server process also runs under Windows

NEW: there is a container for the server that can be used from outside so that the server works directly in its own process and thus runs independently of the main process. This container takes care of providing some information about the server, such as its port and IP address.

UPDATED: The GitHub action for the tests now runs on every pull request and without the need for syntax checks.

UPDATED; The tests for the server have been adjusted so that they work with the other changes.

JanNiklasWeder committed:
Continue test if another fails

dkoch2 commented:
Ich hab in der Start mal statt Server einen ServerContainer erstellt.
Das Spiel startet, aber initialisiert 3 mal Pygame.

multiprocessing.set_start_method("spawn")

![Bildschirmfoto vom 2023-06-09 10-31-31](https://github.com/NAMlab/PlantEd/assets/20724556/a3e04f65-c48a-4e0b-b222-6050ee0b2ffc)

JanNiklasWeder committed:
Merge remote-tracking branch 'origin/buildablestate' into fix/windows

JanNiklasWeder committed:
HOTFIX: Multiple Windows

UPDATED: Moved temperature and water concentration settings to Dynamic Model to avoid multiple windows.

UPDATED: Loop creation of the server adapted so that it can be used without its own process.

UPDATED: Unnecessary imports removed.

UPDATED: Start.py now uses the ServerContainer instead of a server object.
